pkg/flow/worker/worker_pool.go

```diff
@@ -0,0 +1,112 @@
+package worker
+
+import (
+	"fmt"
+	"hash/fnv"
+	"math/rand"
+	"runtime"
+	"sync"
+)
+
+type Pool interface {
```

Comment:
This worker_pool implementation is very generic, wasnt there an available solution that would fit our use case instead of implementing a custom one? This one for example: https://pkg.go.dev/github.com/gammazero/workerpool ?

Reply:
This pool is actually not a typical, generic pool, there are some subtle but important differences. For example, the sharding of the tasks by the key or the fact that the queue is a set, allowing only one instance of given key to be queued at a time. Even the fact that we dequeue just before executing a task will make a difference and would break the tests.

Reply:
got it, thanks!